Trying to write to SD for GPX export leads to crash

I have seen this crash on Google Play.
It seems the user set the GPX export dir to `.../sdcard/gpx/` where I assume we potentially have no write access.
Had no time to try to reproduce it yet, but will do later on.

If this is a general problem, we should either take care, that the user cannot even select this as export dir or we should present an appropriate error message.

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:318)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:762)
Caused by: java.lang.RuntimeException: Aborting on Log.e()
	at cgeo.geocaching.utils.Log.e(Log.java:108)
	at cgeo.geocaching.export.GpxExport$ExportTask.doInBackgroundInternal(GpxExport.java:152)
	at cgeo.geocaching.export.GpxExport$ExportTask.doInBackgroundInternal(GpxExport.java:110)
	at cgeo.geocaching.utils.AsyncTaskWithProgress.doInBackground(AsyncTaskWithProgress.java:104)
	at android.os.AsyncTask$2.call(AsyncTask.java:304)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 4 more
Caused by: java.io.FileNotFoundException: /storage/sdcard0/gpx/export_2017-05-12.gpx (No such file or directory)
	at java.io.FileOutputStream.open(Native Method)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:221)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:169)
	at cgeo.geocaching.export.GpxExport$ExportTask.doInBackgroundInternal(GpxExport.java:143)
	... 8 more
```

--- a/main/src/cgeo/geocaching/storage/LocalStorage.java
+++ b/main/src/cgeo/geocaching/storage/LocalStorage.java
@@ -272,7 +272,12 @@ public final class LocalStorage {
 
     @NonNull
     public static File getGpxExportDirectory() {
-        return new File(Settings.getGpxExportDir());
+        final File gpxExportDir =  new File(Settings.getGpxExportDir());
+        FileUtils.mkdirs(gpxExportDir);
+        if (!gpxExportDir.isDirectory() || !gpxExportDir.canWrite()) {
+            return getDefaultGpxDirectory();
+        }
+        return gpxExportDir;
     }
 
     @NonNull

